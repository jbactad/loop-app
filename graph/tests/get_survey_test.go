package tests

import (
	"testing"

	"github.com/99designs/gqlgen/client"
	"github.com/gkampitakis/go-snaps/snaps"
	"github.com/jbactad/loop/application/ports/mocks"
	"github.com/stretchr/testify/assert"
)

func TestSurvey(t *testing.T) {
	type args struct {
		query string
		id    string
	}
	testCases := []struct {
		name    string
		args    args
		wantErr assert.ErrorAssertionFunc
	}{
		{
			name: "given valid query, should return surveys without error",
			args: args{
				query: `query Survey($surveyId: ID!) {
					survey(id: $surveyId) {
						id
						name
						description
						question
						createdAt
						updatedAt
					}
				}`,
				id: "8138abe3-06b1-5882-88dd-ba931b388f1d",
			},
			wantErr: assert.NoError,
		},
		{
			name: "given invalid query, should return error",
			args: args{
				query: `query Survey($surveyId: ID!) {
					survey(id: $surveyId) {
						id
						name
						description
						question
						createdAt
						updatedAt
					}
				}`,
				id: "",
			},
			wantErr: func(tt assert.TestingT, err error, i ...interface{}) bool {
				assert.Error(tt, err)
				assert.Contains(tt, err.Error(), "graphql: missing required query variable")
				return true
			},
		},
	}
	for _, tt := range testCases {
		t.Run(tt.name, func(t *testing.T) {
			c := NewTestClient(t, mocks.NewTimeProvider(t), mocks.NewUUIDGenerator(t))

			var resp struct {
				Survey struct {
					ID          string
					Description string
					Name        string
					Question    string
					CreatedAt   string
					UpdatedAt   string
				}
			}

			err := c.Post(tt.args.query,
				&resp,
				client.Var("surveyId", tt.args.id),
			)
			if !tt.wantErr(t, err) {
				return
			}

			snaps.MatchJSON(t, resp)
		})
	}
}
