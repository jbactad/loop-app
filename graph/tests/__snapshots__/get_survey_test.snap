
[TestSurvey/given_valid_query,_should_return_surveys_without_error - 1]
{
 "Survey": {
  "CreatedAt": "2019-01-01T00:00:00Z",
  "Description": "Survey 1 description",
  "ID": "8138abe3-06b1-5882-88dd-ba931b388f1d",
  "Name": "Survey 1",
  "Question": "What is your favorite color?",
  "UpdatedAt": "2019-02-01T00:00:00Z"
 }
}
---

[TestSurvey/given_invalid_query,_should_return_error - 1]
{
 "Survey": {
  "CreatedAt": "",
  "Description": "",
  "ID": "",
  "Name": "",
  "Question": "",
  "UpdatedAt": ""
 }
}
---
